packages/utils/parcel-watcher-watchman-js/src/wrapper.js

```diff
@@ -0,0 +1,181 @@
+// @flow
+import * as fs from 'fs';
+import * as path from 'path';
+import * as watchman from 'fb-watchman';
+import {isGlob} from '@parcel/utils';
+import type {
+  Options,
+  Event,
+  SubscribeCallback,
+  AsyncSubscription,
+} from '@parcel/watcher';
+
+type FilePath = string;
+type GlobPattern = string;
+
+// Matches the Watcher API from "@parcel/watcher"
+export interface Watcher {
+  getEventsSince(
+    dir: FilePath,
+    snapshot: FilePath,
+    opts?: Options,
+  ): Promise<Array<Event>>;
+  subscribe(
+    dir: FilePath,
+    fn: SubscribeCallback,
+    opts?: Options,
+  ): Promise<AsyncSubscription>;
+  unsubscribe(
+    dir: FilePath,
+    fn: SubscribeCallback,
+    opts?: Options,
+  ): Promise<void>;
+  writeSnapshot(
+    dir: FilePath,
+    snapshot: FilePath,
+    opts?: Options,
+  ): Promise<FilePath>;
+}
+
+let client: watchman.Client | null = null;
+function getClient() {
```

Comment:
Can't we just init this in the `ParcelWatcherWatchmanJS` constructor?

Reply:
I can refactor a bit, honestly I just left this as I inherited it.